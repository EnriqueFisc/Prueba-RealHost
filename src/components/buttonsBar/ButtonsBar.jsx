import React from "react";

export const ButtonsBar = ({ selectWallFunc }) => {
  return (
    <div className="buttonsBar__container">
      <button onClick={() => selectWallFunc(1)}>Pared 1</button>
      <button onClick={() => selectWallFunc(2)}>Pared 2</button>
      <button onClick={() => selectWallFunc(3)}>Pared 3</button>
      <button onClick={() => selectWallFunc(4)}>Pared 4</button>
    </div>
  );
};
