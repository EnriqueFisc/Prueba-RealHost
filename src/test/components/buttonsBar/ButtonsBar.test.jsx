import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { ButtonsBar } from "../../../components/buttonsBar/ButtonsBar";

describe("Test in ButtonsBar Component", () => {
  test("should render <ButtonsBar/>", () => {
    render(<ButtonsBar />);
    expect(screen.getByText("Pared 1")).toBeDefined();
    expect(screen.getByText("Pared 2")).toBeDefined();
    expect(screen.getByText("Pared 3")).toBeDefined();
    expect(screen.getByText("Pared 4")).toBeDefined();
  });
  test("should match with snapshot", () => {
    const component = render(<ButtonsBar />);
    expect(component.container).toMatchSnapshot();
  });
});
