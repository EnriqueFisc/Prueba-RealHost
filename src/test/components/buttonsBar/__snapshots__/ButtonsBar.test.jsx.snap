// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test in ButtonsBar Component > should match whit snapshot 1`] = `
<div>
  <div
    class="buttonsBar__container"
  >
    <button>
      Pared 1
    </button>
    <button>
      Pared 2
    </button>
    <button>
      Pared 3
    </button>
    <button>
      Pared 4
    </button>
  </div>
</div>
`;

exports[`Test in ButtonsBar Component > should match with snapshot 1`] = `
<div>
  <div
    class="buttonsBar__container"
  >
    <button>
      Pared 1
    </button>
    <button>
      Pared 2
    </button>
    <button>
      Pared 3
    </button>
    <button>
      Pared 4
    </button>
  </div>
</div>
`;
